connector/failoverconnector/README.md

```diff
@@ -0,0 +1,79 @@
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
+
+[in development]:https://github.com/open-telemetry/opentelemetry-collector#in-development
+[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
+
+## Supported Pipeline Types
+
+| [Exporter Pipeline Type] | [Receiver Pipeline Type] | [Stability Level] |
+| ------------------------ | ------------------------ |-------------------|
+| traces | traces | [In development]  |
+| metrics | metrics | [In development]           |
+| logs | logs | [In development]           |
+
+[Exporter Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#exporter-pipeline-type
+[Receiver Pipeline Type]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/connector/README.md#receiver-pipeline-type
+[Stability Level]: https://github.com/open-telemetry/opentelemetry-collector#stability-levels
+<!-- end autogenerated section -->
+
+Allows for health based routing between trace, metric, and log pipelines depending on the health of target downstream exporters.
+
+## Configuration
+
+If you are not already familiar with connectors, you may find it helpful to first visit the [Connectors README].
+
+The following settings are available:
+
+- `priority (required)`: list of pipeline level priorities in a 1 - n configuration, multiple pipelines can sit at a single priority level.
+- `retry_interval (required)`: the frequency at which the pipeline levels will attempt to reestablish connection with all higher priority levels
+- `retry_gap (required)`: the amount of time between trying two separate priority levels in a single retry_interval timeframe
+- `max_retry (required)`: the maximum retries per level
+
+Example:
+
+```yaml
+connectors:
+  failover:
+    priority:
+      - [traces/first, traces/fourth]
+      - [traces/second]
+      - [traces/third]
+    retry_pipeline_interval: 5m
+    retry_gap: 1m
```

Comment:
Is "interval" accurate for how you expect this timing to behave? We use this term in a lot of similar situations, so I'd suggest it here as well unless there's a nuance to the behavior which makes it inaccurate.
```suggestion
    retry_interval: 1m
```

Reply:
So the interval is the time between the cycles of starting over from the highest priority pipeline down to the current stable one while gap is how long it waits between two levels in a single interval’s timeframe. If there’s a naming convention that aligns better to other similar uses, I’m all for it.

Reply:
Hmm, what's the reason for needing to add a delay here? Why not just immediately sent to the next priority after we drop down a level?

Reply:
The idea is that it should retry a maximum of one unhealthy level before returning to the stable level to avoid back pressure through the collector. Especially with retries enabled on the exporters it could take several seconds for each level that is retried. If we’re on a low enough level and each higher priority level needs to be retried prior to returning to stability, that timeframe could become significant very quickly.

Reply:
@djaglowski, any thoughts?

Reply:
Ok, if I'm understanding correctly, any level which fails a retry will immediately end all retries until the next interval. However, if a retry is successful, we will both 1) reset the current level to the successful level, and 2) continue to retry any lower levels - and this gap is the time we wait until we retry the next level?

Reply:
@djaglowski  Let me give an example:

connectors:
  failover:
    priority:
      - [traces/first]
      - [traces/second]
      - [traces/third]
      - [traces/fourth]
    retry_interval: 10m
    retry_gap: 1m
    max_retry: 10

Lets say we are currently on level 4 on the `priority_level` list. At the start of the retry_interval, the connector will try to reestablish the pipeline on level 1 (trace/first). If it fails, the connector will return to level 4 (traces/fourth) and wait the 1m as the retry_gap, when that 1m passes it will now retry level 2 (traces/second) and if that fails will first return to level 4 before waiting another 1m until trying level 3. Once it tries level 3 and it fails, it will return to level 4 and wait the 10m retry_interval again before repeating the process.

The idea for the retry_gap and returning to the _stable level_ between the _retry level_ is to make sure that the connector has a chance to export data without introducing a significant time gap or significant backpressure.

The idea for setting a retry_interval is to avoid constant retries, and allow for whatever needs to be resolved downstream to be resolved (e.g., a collector gateway that is down)

To answer you other question, if a retry is successful then yes the retried level becomes the stable level, and the connector will continue to retry any lower levels that haven't exceeded the `max_retries`

Reply:
Thanks for clarifying @akats7. It seems worth capturing more detail in the readme description. Perhaps this example would be useful as a standalone section and then you can refer to it from the short descriptions in the table.

Reply:
Sure thing, will add that to the readme.